File: perf/benchmarks/micro/sort/sort.go
// Sort N integers — stdlib-sort micro-benchmark (Go).
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"time"
)

func main() {
	n := int64(2_000_000)
	if len(os.Args) > 1 {
		v, _ := strconv.ParseInt(os.Args[1], 10, 64)
		n = v
	}
	arr := make([]int64, n)
	seed := int64(12345)
	for i := int64(0); i < n; i++ {
		seed = (seed*1103515245 + 12345) & 0x7FFFFFFF
		arr[i] = seed
	}
	t0 := time.Now()
	sort.Slice(arr, func(i, j int) bool { return arr[i] < arr[j] })
	ms := float64(time.Since(t0).Nanoseconds()) / 1_000_000.0
	cs := arr[0] ^ arr[n/2] ^ arr[n-1]
	fmt.Printf("CHECKSUM:%d\n", cs)
	fmt.Printf("ELAPSED_MS:%.6f\n", ms)
}

File: perf/benchmarks/micro/sort/sort.rs
// Sort N integers — stdlib-sort micro-benchmark (Rust).
use std::env;
use std::time::Instant;

fn main() {
    let args: Vec<String> = env::args().collect();
    let n: i64 = if args.len() > 1 { args[1].parse().unwrap_or(2_000_000) } else { 2_000_000 };
    let mut arr: Vec<i64> = vec![0; n as usize];
    let mut seed: i64 = 12345;
    for i in 0..n as usize {
        seed = (seed.wrapping_mul(1103515245).wrapping_add(12345)) & 0x7FFFFFFF;
        arr[i] = seed;
    }
    let t0 = Instant::now();
    arr.sort();
    let ms = t0.elapsed().as_nanos() as f64 / 1_000_000.0;
    let cs = arr[0] ^ arr[(n / 2) as usize] ^ arr[(n - 1) as usize];
    println!("CHECKSUM:{}", cs);
    println!("ELAPSED_MS:{:.6}", ms);
}
